### PR TITLE
Refactor: addon dependency installation logic

### LINKED ISSUE #5967: [Feature] A proposal to refactor addon dependency installation logic
**Is your feature request related to a problem? Please describe.**
There are a few problems with the current addon dependency installation logic:

1. Locally installed addon will be overwritten by upstream dependent addon installation logic
2. .  In cases where a addon has multiple topologies(such as loki、prometheus-server),  using one of them as the installation clusters of the current addon is not accurate
3.  When a addon(such as loki) that can be installed on a subcluster depends on a addon(such as o11y-definitions) that can only be installed on the local cluster, installing the loki addon may change the clusters parameter of the o11y-definitions addon, which makes no sense since the o11y-definitions addon itself does not have a clusters parameter

**Describe the solution you'd like**
1. In the first case, add a check whether it is a local install, and if it is a local install, don't override it.
2. In the second case, change the addon's clusters parameter to get it from the secret file
3. In the third case,  add a test of whether the addon has a clusters parameterr

To summarize, the addon dependency installation logic will been refactored as follows:
image

**Describe alternatives you've considered**
For the third case, the method of modifying the topology policy of the application corresponding to the addon was considered, but when combing all the addon definitions in catalog repository, it was found that there was a case that a addon had more than one topology policy, so the method could not be implemented.

**Additional context**

second case:
image
third case:
image

### PR DESCRIPTION
**Pull request by zhaohuiweixiao**
### Summary
🔧🧹➕

Refactor and enhance addon dependency installation logic in `pkg/addon/addon.go`. Simplify function return value, fix parameter bug, use more robust function, and remove redundant function.

> _`install_deps` changed_
> _simpler, robust, bug-free_
> _autumn of cleanup_

### Walkthrough
*  Simplify the return value of `checkDependencyNeedInstall` by removing `depClusters` ([link](https://github.com/kubevela/kubevela/pull/6045/files?diff=unified&w=0#diff-dfc3f6e3b7231b7b41cf82b8e561f3cbf8191d214886e21d798f287f693806dbL1005-R1005))
*  Fix a bug in `installDependency` by passing the correct `addonClusters` to `getDependencyArgs` ([link](https://github.com/kubevela/kubevela/pull/6045/files?diff=unified&w=0#diff-dfc3f6e3b7231b7b41cf82b8e561f3cbf8191d214886e21d798f287f693806dbL1019-R1019))
*  Replace `checkDependencyNeedInstall` with a new function `checkDependencyAddonNeedInstall` that has more comprehensive and accurate logic ([link](https://github.com/kubevela/kubevela/pull/6045/files?diff=unified&w=0#diff-dfc3f6e3b7231b7b41cf82b8e561f3cbf8191d214886e21d798f287f693806dbL1075-R1198))
*  Remove `getDependencyArgs` as it is no longer needed ([link](https://github.com/kubevela/kubevela/pull/6045/files?diff=unified&w=0#diff-dfc3f6e3b7231b7b41cf82b8e561f3cbf8191d214886e21d798f287f693806dbL1117-R1210))

Fixes #5967

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested
`vela addon enable rollout`
`vela addon enable rollout --clusters={lcoal, devops}`
```
vela addon enable ./fluxcd --clusters={local}
vela addon enable rollout --clusters={local, devops}
```

### Special notes for your reviewer

@wangyikewxgm

### CODE REVIEW

File: e2e/addon/mock/testdata/mock-dependence/parameter.cue [hunk 1]
```diff
@@ -7,6 +7,6 @@
 parameter: {
 	// +usage=Custom parameter description
 	myparam: *"myns" | string
-        //+usage=Deploy to specified clusters. Leave empty to deploy to all clusters.
+  //+usage=Deploy to specified clusters. Leave empty to deploy to all clusters.
```
Comment (wangyikewxgm) [inline e2e/addon/mock/testdata/mock-dependence/parameter.cue:10]: Please align.

File: pkg/addon/addon.go [hunk 2]
```diff
@@ -844,6 +844,18 @@ func RenderArgsSecret(addon *InstallPackage, args map[string]interface{}) *unstr
...
 
+// deleteAddonSecret delete the addon's parameter secret
+func deleteAddonSecret(ctx context.Context, k8sClient client.Client, addonName string) {
+	var sec v1.Secret
+	err := k8sClient.Get(ctx, client.ObjectKey{Namespace: types.DefaultKubeVelaNS, Name: addonutil.Addon2SecName(addonName)}, &sec)
+	if err == nil {
+		err = k8sClient.Delete(ctx, &sec)
+		if err != nil {
+			fmt.Printf("fail to delete %v addon parameter file", addonName)
+		}
```
Comment (wangyikewxgm) [inline pkg/addon/addon.go:855]:
```
if err := k8sClient.Get(ctx, client.ObjectKey{Namespace: types.DefaultKubeVelaNS, Name: addonutil.Addon2SecName(addonName)}, &sec); err == nil {
		if err = k8sClient.Delete(ctx, &sec); err != nil {
			fmt.Printf("fail to delete %v addon parameter file", addonName)
		}
......
}
```

We also should consider the situation of if error is `errNotFound`

Comment (zhaohuiweixiao) [reply pkg/addon/addon.go:857]: this code is only for e2e test. the real fix is: https://github.com/kubevela/kubevela/pull/6065(after it was merged, this code will be removed)

File: pkg/addon/helper.go [hunk 3]
```diff
@@ -193,20 +193,22 @@ func FindAddonPackagesDetailFromRegistry(ctx context.Context, k8sClient client.C
 	if len(addonNames) == 0 {
 		return nil, fmt.Errorf("no addon name specified")
 	}
-
+	fmt.Printf("%v addon's installedRegistry: %v\n", addonNames, registryNames)
```
Comment (Somefive) [inline pkg/addon/helper.go:196]: Is this for debug?

File: pkg/addon/addon_suite_test.go [hunk 4]
```diff
@@ -175,72 +194,71 @@ var _ = Describe("Addon test", func() {
...
+		Expect(needInstallAddonDep1).Should(BeFalse())
+
+		// case6: dependency addon has clusters arg, clusters value is local, addonClusters is ["cluster1", "cluster2"]
+		needInstallAddonDep2, err2 := checkDependencyNeedInstall(ctx, k8sClient, hasClusterArgAddonName, addonClusters)
+		Expect(err2).Should(BeNil())
+		Expect(needInstallAddonDep2).Should(BeTrue())
 	})
 
-	It("getDependencyArgs func test", func() {
+	/*It("getDependencyArgs func test", func() {
```
Comment (Somefive) [inline pkg/addon/addon_suite_test.go:261]: If not used, could we remove it?

File: pkg/addon/helper.go (hunk 3 above)
Comment (zhaohuiweixiao) [reply pkg/addon/helper.go:196]: Yes，I remove these.

File: pkg/addon/addon_suite_test.go (hunk 4 above)
Comment (zhaohuiweixiao) [reply pkg/addon/addon_suite_test.go:261]: ok

**Comment by zhaohuiweixiao** [comment]
@Somefive please help to review the https://github.com/kubevela/kubevela/pull/6065, this pr dependents it.

**Review by Somefive — APPROVED** [review]
Looks generally good to me

**Review by wangyikewxgm — APPROVED** [review]

**Pull request merged**